### DISCUSSION TITLE
stdout of running processes?

### DISCUSSION THREAD

**Discussion by magpie514** [hluk/CopyQ#2288] [Q&A]
I've noticed that when running commands from copyq (such as inside an action) there will be no stdout at all. Not from the running programs or from simple "echo/printf" statements in the commands.
I was trying to debug a script called from within copyq when certain content is copied in the clipboard and the lack of stdout collection made it complicated. 

So the question is, is there some way for copyq to output stdout/stderr from the commands it's running? I tried looking into options and command line arguments but I can't seem to find anything (I might have missed something obvious though).

**Comment by hluk** [comment]
In Action dialog, if you set "Store standard output" to "text/plain", the stdout of the process will be stored in a new item.

**Reply by magpie514** [reply]
Thanks for replying!
Using an item as buffer has uses, I'll definitely use it for something else I have in mind.  But in this case it'd also require extra logic to output the message as it happens and such, and it'd be a bit hackish.
For reference, I unified all my system messages into a processed stream in a terminal, making it very convenient to see warnings or errors as they happen. Under normal circumstances my CopyQ scripts or called tools are supposed to be silent unless there's an error, and was counting on them being logged with everything else, but then I ran into an error condition and took a while to notice, and debugging was complicated without prints to keep track of the internal state.

So I assume stdout is always intercepted and there's no way to do it currently without editing copyq itself? I have a couple alternate solutions, so it'll be fine either way, but just to be sure.
